tools/ts-generator/rpc/rpc_gen_js.py

```diff
@@ -79,72 +82,135 @@
     RpcHeader.toBytes(rpc, reserve);
 }
 
+const startReadRequest = (fn: ApplyFn) => (socket: Socket): void => {
+  socket.once("data", (data) => readRequestBuffer(data, fn, socket));
+};
+
+/**
+ * given a buffer with request, it read a rpc header and rpc header payload size
+ * if the buffer has a complete payload size it return buffer, Otherwise, it
+ * waits for n "data" events with necessary data for complete payload size.
+ * @param buffer
+ * @param fn
+ * @param socket
+ */
+const readRequestBuffer = (buffer: Buffer, fn: ApplyFn, socket: Socket) => {
+  const [rpcHeader, crcValidation, crc32] = validateRpcHeader(
+    buffer.slice(0, rpcHeaderSize)
+  );
+  if (!crcValidation) {
+    throw (
+      `Crc32 inconsistent, expect: ${rpcHeader.headerChecksum} ` +
+      `generated: ${crc32}`
+    );
+  } else {
+    const size = rpcHeader.payloadSize;
+    if (buffer.length - rpcHeaderSize >= size) {
+      const result = buffer.slice(rpcHeaderSize, size + rpcHeaderSize);
+      startReadRequest(fn)(socket)
+      return fn(rpcHeader, result, socket);
+    } else {
+      const bytesForReading = size - (buffer.length - rpcHeaderSize);
+      return readNextChunk(socket, bytesForReading, fn)
+        .then((nextBuffer) =>
+          Buffer.concat([buffer.slice(rpcHeaderSize), nextBuffer])
+        )
+        .then((result) => fn(rpcHeader, result, socket));
+    }
+  }
+};
+
+/**
+ * given payload size, it waits for n "data" events until have the complete data
+ * size.
+ * @param socket
+ * @param size
+ * @param fn
+ */
+const readNextChunk = (
+  socket: Socket,
+  size: number,
+  fn: ApplyFn
+): Promise<Buffer> => {
+  return new Promise<Buffer>((resolve) => {
+    socket.once("data", (data) => {
+      if (data.length >= size) {
+        if (data.length - size > 0) {
```

Comment:
You're checking if data.length >= size, and then inside that if block you're checking if data.length > size or if they're equal. Seems to me you could get rid of the >= check:
 
```js
if (data.length > size) {
// ...
} else if (data.length == size) {
// ...
} else {
  return readNextChunk(/**/)
// ... 
}
```

Reply:
Umm no, the first conditional says `data. length >= size` says the loop should be finishing, but the second condition `data.length - size *> 0*` says if the data contains another request. in that case, it passes an extra request to `readBufferRequest`, otherwise, it calls `startReadRequest` in order to wait for a new request. @0x5d

Reply:
Done